instrumentation/micrometer/micrometer-1.5/testing/src/main/java/io/opentelemetry/instrumentation/micrometer/v1_5/AbstractTimerTest.java

```diff
@@ -197,4 +198,49 @@ void testMicrometerPercentiles() {
                                 point ->
                                     assertThat(point).attributes().containsEntry("phi", "0.99"))));
   }
+
+  @Test
+  void testMicrometerMax() throws InterruptedException {
+    // given
+    Timer timer =
+        Timer.builder("testTimerMax")
+            .description("This is a test timer")
+            .tags("tag", "value")
+            .register(Metrics.globalRegistry);
+
+    // when
+    timer.record(1, TimeUnit.SECONDS);
+    timer.record(2, TimeUnit.SECONDS);
+    timer.record(4, TimeUnit.SECONDS);
+
+    // then
+    testing()
+        .waitAndAssertMetrics(
+            INSTRUMENTATION_NAME,
+            "testTimerMax.max",
+            metrics ->
+                metrics.anySatisfy(
+                    metric ->
+                        assertThat(metric)
+                            .hasDescription("This is a test timer")
+                            .hasDoubleGauge()
+                            .points()
+                            .anySatisfy(
+                                point ->
+                                    assertThat(point)
+                                        .hasValue(4_000)
+                                        .attributes()
+                                        .containsEntry("tag", "value"))));
+
+    // when
+    Metrics.globalRegistry.remove(timer);
+    Thread.sleep(10); // give time for any inflight metric export to be received
```

Comment:
Will be able to remove this after #5314 I think

Reply:
I'm guessing there may still be a tiny window while the metric sdk is still collecting metrics but before it has sent them to the metric exporter(?)

Reply:
At time of writing, that PR was causing clearData to first flush the metrics SDK. But realized it's not necessary or helpful in most tests so changed it back, so indeed this comment is now invalid

Reply:
oh i c, it flushed the full collection cycle, not just the export? that does seem useful for all of these micrometer unregister tests that we have